Source/Core/DolphinWX/ISOFile.cpp

```diff
@@ -242,41 +242,40 @@ bool GameListItem::SetWxBannerFromPngFile(const std::string& path)
 
 void GameListItem::SetWxBannerFromRaw(const Banner& banner)
 {
+  if (banner.empty())
+    return;
+
   // Need to make explicit copy as wxImage uses reference counting for copies combined with only
   // taking a pointer, not the content, when given a buffer to its constructor.
-  if (!banner.empty())
-  {
-    m_banner_wx.Create(banner.width, banner.height, false);
-    std::memcpy(m_banner_wx.GetData(), banner.buffer.data(), banner.buffer.size());
-  }
+  m_banner_wx.Create(banner.width, banner.height, false);
+  std::memcpy(m_banner_wx.GetData(), banner.buffer.data(), banner.buffer.size());
 }
 
 bool GameListItem::BannerChanged()
 {
   // Wii banners can only be read if there is a savefile,
   // so sometimes caches don't contain banners. Let's check
   // if a banner has become available after the cache was made.
-  if ((m_platform == DiscIO::Platform::WII_DISC || m_platform == DiscIO::Platform::WII_WAD) &&
-      m_banner.empty())
-  {
-    auto& banner = m_pending.banner;
-    std::vector<u32> buffer =
-        DiscIO::Volume::GetWiiBanner(&banner.width, &banner.height, m_title_id);
-    if (buffer.size())
-    {
-      ReadVolumeBanner(&banner.buffer, buffer, banner.width, banner.height);
-      // Only reach here if m_banner was empty, so don't need to explicitly compare to see if
-      // different
-      return true;
-    }
-  }
-  return false;
+
+  if (!m_volume_banner.empty() ||
+      (m_platform != DiscIO::Platform::WII_DISC && m_platform != DiscIO::Platform::WII_WAD))
+    return false;
```

Comment:
Dunno about you guys, but I usually try to break up individual, otherwise unrelated checks into separate ifs.

How about this?
```c++
if (!m_volume_banner.empty())
  return false;
if (!DiscIO::IsWii(m_platform)) // not sure if that one accepts a Platform tho...
  return false;
```

Can be ignored if thats just me ;)